doc/man7/EVP_MD-BLAKE2.pod

```diff
@@ -30,6 +30,23 @@ Known names are "BLAKE2B-512" and "BLAKE2b512".
 This implementation supports the common gettable parameters described
 in L<EVP_MD-common(7)>.
 
+=head2 Settable Context Parameters
+
+These implementations support the following L<OSSL_PARAM(3)> entries,
+settable for an B<EVP_MD_CTX> with L<EVP_MD_CTX_set_params(3)>:
+
+=over 4
+
+=item "xoflen" (B<OSSL_DIGEST_PARAM_XOFLEN>) <unsigned integer>
```

Comment:
I am not convinced that `OSSL_DIGEST_PARAM_XOFLEN>` (_xoflen_) is the correct parameter here because there is no capability for _extension_ of the output.  I.e. it's not really an XOF digest despite having the capability for variable output length.

To me, OSSL_DIGEST_PARAM_SIZE` (_size_) seems more appropriate.

There is work in progress to add proper XOF support which won't work for these.  The [design](https://github.com/openssl/openssl/pull/21443) and [implementation](https://github.com/openssl/openssl/pull/21511) PRs are outstanding.

Reply:
Using OSSL_DIGEST_PARAM_SIZE as settable parameter would be quite weird. And although Blake2 is not really XOF the EVP_DigestFinalXOF is not really XOF function as there is nothing extensible in the current implementation of it and even Shane's PRs linked above do not change anything about that.

The problem is Blake2b is already made variable-length output function in the current master branch but Blake2s isn't which is confusing.